specification/trace/semantic_conventions/messaging.md

```diff
@@ -80,6 +80,48 @@ Often such destinations are unnamed or have an auto-generated name.
 
 Given these definitions, the remainder of this section describes the semantic conventions for Spans describing interactions with messaging systems.
 
+### Context propagation
+
+A message may pass many different components and layers in one or more intermediaries
+when it is propagated from the producer to the consumer(s). To be able to correlate
+consumer traces with producer traces using the existing context propagation mechanisms,
+all components must propagate context down the chain.
+
+This imposes a difficulty because it cannot be assumed, and in many cases,
+it is not even desired, that all components and layers to be instrumented and
+propagate context according to OpenTelemetry requirements.
+
+To be able to correlate consumer traces with producer traces without requiring
+intermediary instrumentation, the context needs to be propagated on a
+*per-message* basis instead of on a *per-request* basis. This allows all components
+to have access to the same per-message context information, making it possible
+to correlate all the stages involved in processing a message with the message's
+creation.
+
+A message *creation context* allows correlating producer with consumer(s)
+of a message and model the dependencies between them,
+regardless of the underlying messaging transport mechanism and its instrumentation.
+
+The message creation context is created by the producer and should be propagated
+to the consumer(s). Producer and consumer applications should be instrumented
+in a way so that the creation context is attached to messages and extracted
+from messages in a coordinated way.
```

Comment:
What does that mean? Do you mean a standardized way?

Reply:
I think what the original author of OTEP wanted to say is that producers and consumers must attach/extract the context at certain "points" during the message lifetime, in order to create a meaningful trace. For ex, producer would attach the context before sending, and a consumer would extract it upon receiving.. thus in a "coordinated" way.

Reply:
I think I will remove this section: "Producer and consumer applications should be instrumented in a way [...]". It doesn't add any value and the exact way this will work is what the group is working on to define right now.

Reply:
Done [6c0c7c7](https://github.com/dynatrace-oss-contrib/opentelemetry-specification/pull/22/commits/6c0c7c7b87a932d080561174c0a3141642373953)